modules/dictionary.lobster

```diff
@@ -35,15 +36,37 @@ class dictionary<K,V>:
         if c:
             c.value = value
         else:
+            sz++
             buckets[h] = chain<K,V> { buckets[h], key, value }
 
+    def del(key:K):
+        let h = hash(key) % buckets.length
+        var c = buckets[h]
+        var prev = nil::chain<K,V>
+        while c:
+            if c.key == key:
+                break
+            prev = c
+            c = c.next
+        guard c
+        sz--
+        if c == buckets[h]:
+            buckets[h] = c.next
+        else:
+            guard prev
+            prev.next = c.next
+
+    def size():
```

Comment:
what is this member/function useful for? typically a size is only useful in combination with iterating all elements, for which we so far only have `get_all`

Reply:
Initially I thought it would be convenient to use like std::set, e.g. for counting unique records. On second thought, keys will be eventually checked after that. So yeah, maybe it's a redundant method indeed.